integration-tests/cucumber-tests-platform-smartmetering/src/test/java/org/opensmartgridplatform/cucumber/platform/smartmetering/support/ws/smartmetering/configuration/SetEncryptionKeyExchangeOnGMeterRequestFactory.java

```diff
@@ -26,6 +28,19 @@ public static SetEncryptionKeyExchangeOnGMeterRequest fromParameterMap(
     setEncryptionKeyExchangeOnGMeterRequest.setDeviceIdentification(
         requestParameters.get(PlatformSmartmeteringKeys.KEY_DEVICE_IDENTIFICATION));
 
+    final SetEncryptionKeyExchangeOnGMeterRequestData data =
+        new SetEncryptionKeyExchangeOnGMeterRequestData();
+
+    data.setMbusDeviceIdentification(
+        requestParameters.get(PlatformSmartmeteringKeys.KEY_DEVICE_IDENTIFICATION));
+    data.setSecretType(
+        SecretType.valueOf(requestParameters.get(PlatformSmartmeteringKeys.SECRET_TYPE)));
+    data.setCloseOpticalPort(
+        Boolean.valueOf(
+            requestParameters.getOrDefault(PlatformSmartmeteringKeys.CLOSE_OPTICAL_PORT, "false")));
```

Comment:
Hoort "false" niet in de PlatformSmartmeteringDefaults thuis?
 _PlatformSmartmeteringDefaults.CLOSE_OPTICAL_PORT_

Reply:
yes, PlatformSmartmeteringDefaults.CLOSE_OPTICAL_PORT is added.